src/main/java/com/amazon/dlic/auth/http/saml/SamlDOMMetadataResolver.java

```diff
@@ -0,0 +1,55 @@
+package com.amazon.dlic.auth.http.saml;
+
+import net.shibboleth.utilities.java.support.component.ComponentInitializationException;
+import net.shibboleth.utilities.java.support.xml.BasicParserPool;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.settings.Settings;
+import org.opensaml.saml.metadata.resolver.impl.DOMMetadataResolver;
+
+import java.io.IOException;
+import java.io.StringReader;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+public class SamlDOMMetadataResolver extends DOMMetadataResolver {
+    protected final static Logger log = LogManager.getLogger(SamlDOMMetadataResolver.class);
+
+    private static int componentIdCounter = 0;
+
+    public SamlDOMMetadataResolver(Settings settings) throws Exception {
+        super(getMetadataDOM(settings));
+        setId(HTTPSamlAuthenticator.class.getName() + "_" + (++componentIdCounter));
```

Comment:
Does this ID needs to be different every time it is initialized?

Reply:
Was following the format of https://github.com/opendistro-for-elasticsearch/security/blob/master/src/main/java/com/amazon/dlic/auth/http/saml/SamlHTTPMetadataResolver.java#L40

Reply:
Consider moving common initialization to `createMetadataResolver()`